Automattic/jetpack
# Classic Theme Helper: Adding in Responsive Video class initialization from the Jetpack Mu WPcom package

Fixes https://github.com/Automattic/vulcan/issues/386

## Proposed changes:

* This PR adds back the Classic Theme Helper package Main class initialization from the Jetpack Mu WPcom package, as it was initially added in this PR - https://github.com/Automattic/jetpack/pull/37969 - and then removed as it's addition would cause fatal errors (functions in the module file being redeclared) if wpcomsh was updated before Jetpack.

### Other information:


## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/386


## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* For this we're primarily testing on Simple and WoA. Since Jetpack is now updated on WoA and Simple sites, we're basically testing with the current Jetpack version and with this PR checked out (using the script in the comment below on Simple, and with this PR applied using the Beta Tester plugin on a WoA site, for the WPcom Features plugin).
* Once the environment is set up, follow the remaining testing instructions here but just for Responsive Videos: https://github.com/Automattic/jetpack/pull/37969
* The script in the page source on WoA sites - `jetpack-responsive-videos-js` -  should be via the `jetpack-mu-wpcom` plugin. Without this PR applied it is the same, and this is because of the new `class_exists` check via the Jetpack module file, which actually ends up running the class file (which has an initialization call) via the Classic Theme Helper package. This will be removed in a later version.
* The script in the page source on Simple sites - `jetpack-responsive-videos-js` - should also be via `jetpack-mu-wpcom` plugin. Without this PR applied it should be via the `responsive-videos.js` script (no ID) if testing whilst not sandboxed, but still `jetpack-responsive-videos-js` if sandboxed without the changes applied (also via the `jetpack-mu-wpcom` plugin, but this is also because of the new `class_exists` check via the Jetpack module file).
* Make sure there are no errors reported - on Simple `tail -f /tmp/php-errors`, on WoA in the debug.log.
